### ISSUE TITLE
the dash pattern cannot be modified

### ISSUE THREAD

**Issue by philiparvidsson** [alpaker/fill-column-indicator#73]
I ran into the problem with line-spacing (I use a line-spacing of 5) where there are spaces between the column guide line segments. I'm thinking I could use a dashed column guide instead, with a line-height of say, 0.25. However, I would like to be able to use two or even tree dashes per line, to make the spacing tighter. Would this be possible?

**Comment by alpaker** [comment]
You can set the options `fci-dash-pattern` to any value between 0 and 1.  It controls the length of the dash, as a fraction of the total line height.

**Comment by philiparvidsson** [comment]
I think you misunderstood. What I wanted to do was, for example, TWO dashes per line. Ie, two 25% long dashes on a single line.

**Comment by alpaker** [comment]
Ok.  Does the following patch approximate what you're looking for?

```
diff --git a/fill-column-indicator.el b/fill-column-indicator.el
index cbe20d4..20dc9c3 100644
--- a/fill-column-indicator.el
+++ b/fill-column-indicator.el
@@ -619,13 +619,14 @@ rough heuristic.)"
     (setq fci-char-width (frame-char-width frame)
           fci-char-height (frame-char-height frame))))

+(defconst fci-mid-margin 2)
 (defmacro fci-with-rule-parameters (&rest body)
   "Define various quantites used in generating rule image descriptors."
   (declare (indent defun))
   `(let* ((height-str (number-to-string fci-char-height))
           (width-str (number-to-string fci-char-width))
           (rule-width (min fci-rule-width fci-char-width))
-          (hmargin (/ (- fci-char-width rule-width) 2.0))
+          (hmargin (/ (- fci-char-width rule-width rule-width fci-mid-margin) 2.0))
           (left-margin (floor hmargin))
           (right-margin (ceiling hmargin))
           (segment-ratio (if fci-rule-use-dashes fci-dash-pattern 1))
@@ -670,6 +671,8 @@ rough heuristic.)"
            (on-pixels (concat "\""
                               (make-string left-margin ?0)
                               (make-string rule-width ?1)
+                              (make-string fci-mid-margin ?0)
+                              (make-string rule-width ?1)
                               (make-string right-margin ?0)
                               "\","))
            (off-pixels (concat "\"" (make-string fci-char-width ?0) "\","))
```

**Comment by philiparvidsson** [comment]
I'm not versed enough in elisp to tell—I'm sorry. To be concise: What I want to do is have more than one segment per line. Reason: I could have a dotted column guide by having 3-4 segments per line, setting their ratio to 0.1.

**Comment by alpaker** [comment]
I just meant to apply the patch and try out the resulting behavior.  

I've pushed a branch`multi-line-test`.  Please give that a try.

**Comment by philiparvidsson** [comment]
No, your patch provides multiple lines per line in a horizontal fashion. I want multiple segments per line along the vertical axis. See mockup below:

https://snag.gy/RPIwsS.jpg

Does this make it clear what I mean?

**Comment by alpaker** [comment]
That's not currently possible, but it would be straightforward to implement.  

I'm not sure what the best configuration API is.  Using fractions of line height seems awkward for this sort of pattern.  I think it'd be better to have two options:  one that specifies line segment length, and another that specifies the gap between segments.  I'm entirely open to suggestions, though.  What do you think?

**Comment by philiparvidsson** [comment]
I agree that fractions is an awkward way to configure it. I think, for this kind of dash pattern, you would want to specify two variables; one for the length of the dashes/lines, and one for the gap between them, as you say. I'm thinking it would be best to configure their length as number-of-pixels here.

i.e:

(setq fci-dash-pattern '(x y))

Where x is the number of pixels to draw, and y is the number of pixels to skip (before drawing x pixels again, and repeat...)

**Comment by alpaker** [comment]
Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
